src/telemetry.rs

```diff
@@ -100,11 +96,28 @@ impl Telemetry {
 
         if self.sentry_guard.is_some() {
             sentry::capture_error(error);
+        } else if let Some(posthog_client) = &self.posthog {
+            let mut exception = Exception::new(error, &self.config.instance_id);
+            Telemetry::add_default_props(&mut exception)?;
+
+            posthog_client.capture_exception(exception)
+                .map_err(|e| TelemetryError::SendError(e.to_string()))?;
         }
 
         Ok(())
     }
 
+    fn add_default_props(event: &mut impl EventBase) -> TelemetryResult<()> {
+        // Add default properties
+        event.insert_prop("platform", std::env::consts::OS)
+            .map_err(|e| TelemetryError::SendError(e.to_string()))?;
+
+        event.insert_prop("version", env!("CARGO_PKG_VERSION"))
```

Comment:
`CARGO_PKG_VERSION` corresponds to the version of the package being built, e.g. in this case -- version of the library.
Not sure if that's intended. If you want to log the binary version, it should be provided to the `Telemetry` object as an argument.

Reply:
This is meant to be the telemetry lib version.

Reply:
Just for us to know which version of lib reported a particular event

Reply:
Maybe name it as a `zksync_telemetry_version` so that it's not confused?

Reply:
Agreed

Reply:
Done, also added app name and app version to default props, had to [change function pointer to closure](https://github.com/Romsters/posthog-rs/commit/8b228083d0c8d44826024882a47b02b70ca94229)